cmake/KokkosConfig.cmake.in

```diff
@@ -69,4 +69,13 @@ elseif(@Kokkos_ENABLE_CUDA@ AND NOT @KOKKOS_COMPILE_LANGUAGE@ STREQUAL CUDA AND
   unset(IS_NVCC)
 endif()
 
+if(@Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE@)
+  if(NOT (${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES} STREQUAL ${Kokkos_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}))
+    message(
+      WARNING
+        "Kokkos_ENABLE_ARCH_... is not the same architecture as CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES. This can lead to problems when Kokkos and @KOKKOS_COMPILE_LANGUAGE@ code is mixed."
```

Comment:
```suggestion
        "Kokkos_ARCH_... is not the same architecture as CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES. This can lead to problems when Kokkos and @KOKKOS_COMPILE_LANGUAGE@ code is mixed."
```
or preferably showing the actual values:
```suggestion
        "Kokkos enabled the architecture '${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}' which is not the same the  is not the same architecture as the architectures enabled in CMake "${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}'. This can lead to problems when Kokkos and @KOKKOS_COMPILE_LANGUAGE@ code is mixed."
```